flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/TableFactoryService.java

```diff
@@ -269,16 +271,67 @@
 			plainContext.remove(CATALOG_PROPERTY_VERSION);
 
 			// check if required context is met
-			return plainContext.keySet()
-				.stream()
-				.allMatch(e -> properties.containsKey(e) && properties.get(e).equals(plainContext.get(e)));
-		}).collect(Collectors.toList());
+			int matchedSize = 0;
+			Map<String, Tuple2<String, String>> unMatchedProperties = new HashMap<>();
+			for (Map.Entry<String, String> e : plainContext.entrySet()) {
+				if (properties.containsKey(e.getKey()) &&
+						properties.get(e.getKey()).equals(e.getValue())) {
+					matchedSize++;
```

Comment:
If we have `unMatchedProperties`, we don't need to track `matchedSize` any more.

Reply:
No, they are different size.
For example, there are two factories: A with 100 properties and B with 10 properties.
Maybe we matched 2 for A and 8 for B, we need choose B, but `unMatchedProperties` of A will be more.
Of course we can check value of `unMatchedProperties`, but I don't want to make it complex.

Reply:
The `matchedSize` can be calculated by `userProperties.size() - unMatchedProperties.size()`.

Reply:
The `matchedSize` can be calculated by `requiredProperties.size() - unMatchedProperties.size()`. Is that right?

Reply:
I got what are mean, we need split `unMatchedProperties` to `missingProperties` and `mismatchedProperties`.

Reply:
Unfortunately, we still don't know, you can take a look to my above example.

Reply:
You can take a look again after updating the code.